feat: Add tox configuration for OpenStack standard testing

paikend opened the pull request:
## Summary
Add tox configuration following OpenStack standards while keeping uv for development.

## Changes
- Add `tox.ini` with OpenStack-style environments
- Add Development section to README.md with both uv and tox workflows
- Add `.DS_Store` to `.gitignore`

## Motivation
Most OpenStack projects use tox for testing and CI/CD. This allows OpenStack contributors to use familiar tox commands while maintaining uv for fast local development.

## Use Cases
- Run tests: `tox -e py3` or `tox -e py310`
- Run linting: `tox -e pep8`
- Generate coverage: `tox -e cover` (85%+ coverage)
- Run in venv: `tox -e venv -- <command>`
- Auto-format: `tox -e format`

## Testing
All environments tested and working:
- ✅ `tox -e py3` - 156 tests passed
- ✅ `tox -e pep8` - All checks passed
- ✅ `tox -e cover` - 85% coverage
- ✅ `tox -e venv` - Working
- ✅ `uv` workflow - All commands working

#99

paikend committed:
chore: Add .DS_Store to .gitignore

Add macOS system file to gitignore to prevent accidental commits

paikend committed:
feat: Add tox configuration for OpenStack standard testing

Add tox.ini following OpenStack project standards to support:
- Multiple Python versions testing (py3, py310-313)
- Style checking with ruff (pep8 environment)
- Code formatting (format environment)
- Coverage reporting (cover environment)
- Debug support (debug environment)
- Virtual environment for arbitrary commands (venv environment)

This allows OpenStack contributors to use familiar tox commands
while maintaining uv for fast local development.

paikend committed:
docs: Add Development section to README

Add comprehensive development documentation including:
- Setup instructions for both uv and tox workflows
- uv commands for fast local development
- tox commands following OpenStack standards
- Testing section with coverage information (85%+)

This helps contributors choose their preferred development workflow.

paikend committed:
style: Remove trailing whitespace in pyproject.toml

S0okJu reviewed (dismissed):
Thank you for adding .DS_Store.

halucinor commented on tox.ini:
```diff
@@ -0,0 +1,69 @@
+deps =
+    {[testenv]deps}
+    coverage>=7.0
+    pytest-cov>=6.0.0
```
이 의존성 pyproject.toml에도 추가해주실 수 있나요?

paikend replied:
좋습니다! uv 명령어 기준으로 tox를 통합해야겠어요 
어차피 사용자는 인터페이스만 알아도 잘 사용할 것 같네요!

paikend committed:
refactor: Consolidate dependencies in pyproject.toml

  Move tox.ini dependencies to pyproject.toml dependency-groups
  and update all tox environments to use 'uv sync' instead of
  hardcoded dependency lists. This establishes pyproject.toml
  as the single source of truth for dependency management.

  Changes:
  - Add coverage tools to test dependency group
  - Update tox environments to use 'uv sync --group {test|dev}'
  - Replace direct commands with 'uv run' for consistency

  This maintains OpenStack tox standards while leveraging uv's
  speed. All tests passing (156/156, 85% coverage).

halucinor reviewed (approved):
LGTM

S0okJu reviewed (approved):
LGTM
